src/button/Button.tsx

```diff
@@ -6,115 +6,124 @@ import Spinner from "../spinner/Spinner";
 // SCSS import is moved here to be able to override spinner styles without nesting
 import "./_button.scss";
 
-export interface ButtonProps {
-  testid?: string;
+export type ButtonProps = Omit<
+  React.ButtonHTMLAttributes<HTMLButtonElement>,
+  "disabled"
+> & {
   children: React.ReactNode;
+  testid?: string;
   customSpinner?: React.ReactNode;
+  isDisabled?: boolean;
+  shouldPreventDefault?: boolean;
+  shouldStopPropagation?: boolean;
+  shouldFocus?: boolean;
   onClick?: React.ReactEventHandler<HTMLButtonElement>;
   onMouseOver?: React.ReactEventHandler<HTMLButtonElement>;
   onMouseDown?: React.ReactEventHandler<HTMLButtonElement>;
   onMouseUp?: React.ReactEventHandler<HTMLButtonElement>;
   onFocus?: React.ReactEventHandler<HTMLButtonElement>;
   onBlur?: React.ReactEventHandler<HTMLButtonElement>;
   id?: string;
+  ref?: React.RefObject<HTMLButtonElement>;
   lang?: string;
-  type?: "button" | "submit" | "reset";
   shouldDisplaySpinner?: boolean;
-  isDisabled?: boolean;
-  shouldPreventDefault?: boolean;
-  shouldStopPropagation?: boolean;
-  shouldFocus?: boolean;
   ariaLabel?: string;
   customClassName?: string;
   tabIndex?: number;
-}
+};
 
-function Button({
-  testid,
-  type = "button",
-  id,
-  lang,
-  onClick,
-  children,
-  customSpinner,
-  customClassName,
-  shouldPreventDefault = true,
-  shouldStopPropagation = true,
-  shouldFocus,
-  shouldDisplaySpinner,
-  ariaLabel,
-  isDisabled,
-  onMouseOver,
-  onMouseDown,
-  onMouseUp,
-  onFocus,
-  onBlur,
-  tabIndex
-}: ButtonProps) {
-  const isButtonDisabled = Boolean(isDisabled || shouldDisplaySpinner);
-  const containerClassName = classNames("button", customClassName, {
-    "button--is-inactive": isButtonDisabled
-  });
-  const spinnerContent = customSpinner || (
-    <div className={"button__spinner-container"}>
-      <Spinner
-        customClassName={"button__spinner"}
-        aria-label={"Button spinner visible. Button inactivated."}
-      />
-    </div>
-  );
+const Button = React.forwardRef<HTMLButtonElement, Record<string, any>>(
+  // eslint-disable-next-line prefer-arrow-callback
+  function ButtonComponent(props: ButtonProps) {
+    const {
+      testid,
+      type = "button",
+      id,
+      ref,
+      lang,
+      onClick,
+      children,
+      customSpinner,
+      customClassName,
+      shouldPreventDefault = true,
+      shouldStopPropagation = true,
+      shouldFocus,
+      shouldDisplaySpinner,
+      ariaLabel,
+      isDisabled,
+      onMouseOver,
+      onMouseDown,
+      onMouseUp,
+      onFocus,
+      onBlur,
+      tabIndex
```

Comment:
Let's destructure only the values that we actually need and gather the remaining props with a rest operator.
```suggestion
      /* props that we need to use */,
      ...otherProps
```